src/main/java/aquality/appium/mobile/actions/TouchActions.java

```diff
@@ -0,0 +1,45 @@
+package aquality.appium.mobile.actions;
+
+import aquality.appium.mobile.application.AqualityServices;
+import aquality.appium.mobile.configuration.ISwipeConfiguration;
+import aquality.selenium.core.logging.Logger;
+import io.appium.java_client.TouchAction;
+import io.appium.java_client.touch.offset.PointOption;
+import org.openqa.selenium.Point;
+
+import static io.appium.java_client.touch.WaitOptions.waitOptions;
+
+public class TouchActions implements ITouchActions {
+
+    @Override
+    public void swipe(Point startPoint, Point endPoint) {
+        Logger.getInstance().info(String.format(
+                "Swiping: from coordinates x:%s y:%s to x:%s y:%s",
+                startPoint.getX(),
+                startPoint.getY(),
+                endPoint.getX(),
+                endPoint.getY()));
+        getTouchAction().press(PointOption.point(startPoint))
+                .waitAction(waitOptions(AqualityServices.get(ISwipeConfiguration.class).getTimeout()))
+                .moveTo(PointOption.point(endPoint)).release().perform();
+    }
+
+    @Override
+    public void swipeWithLongPress(Point startPoint, Point endPoint) {
+        Logger.getInstance().info(String.format(
+                "Swiping: from coordinates x:%s y:%s to x:%s y:%s",
```

Comment:
1) Please localize your logging value
2) Please move common logic into a private method using UnaryOperator (take a look at "performAction" method here: https://github.com/aquality-automation/aquality-selenium-java/blob/master/src/main/java/aquality/selenium/elements/actions/MouseActions.java)
3) we should probably add a retry for this operation, what do you think about it?

Reply:
1) will do
2) will do
3) will add a number of retries as an argument to the function. By default, there will be 1 try to perform a swipe. Will add both for swipe and swipeWithLongPress.